decam/makeCrosstalkDecam.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""Convert a DECam crosstalk text table into LSST CrosstalkCalibs.
+"""
+import argparse
+import numpy as np
+import os.path
+import sys
+
+import lsst.ip.isr as ipIsr
+from lsst.daf.base import PropertyList
+from lsst.obs.decam import DecamMapper
+
+
+def makeDetectorCrosstalk(dataDict, force=False):
+    """Generate and write CrosstalkCalib from dictionary.
+
+    Parameters
+    ----------
+    dataDict : `dict`
+        Dictionary from ``readFile`` containing crosstalk definition.
+    """
+    dataDict['coeffs'] = dataDict['coeffs'].transpose()
+
+    decamCT = ipIsr.crosstalk.CrosstalkCalib.fromDict(dataDict)
+    # Lie to ingestCuratedCalibs, because it interprets calibDate as validStart.
+    decamCT.updateMetadata(setDate=False, CALIBDATE='1970-01-01T00:00:00')
+
+    detName = dataDict['DETECTOR_NAME']
+    outDir = os.path.join(DecamMapper.getCrosstalkDir(), detName.lower())
+    if os.path.exists(outDir):
+        if not force:
+            print("Output directory %r exists; use --force to replace" % (outDir, ))
+            sys.exit(1)
+    else:
+        os.makedirs(outDir)
+    decamCT.writeText(f"{outDir}/1970-01-01T00:00:00.yaml")
+
+
+def readFile(fromFile):
+    """Construct crosstalk dictionary-of-dictionaries from fromFile.
+
+    Parameters
+    ----------
+    fromFile : `str`
+        File containing crosstalk coefficient information.
+
+    Results
+    -------
+    outDict : `dict` [`str` : `dict`]
+        Output dictionary, keyed on victim detector names, containing
+        `lsst.ip.isr.CrosstalkCalib`'s expected dictionary format.
+
+    Raises
+    ------
+    RuntimeError :
+        Raised if the detector is not known.
+    """
+    ampIndexMap = {'A': 0, 'B': 1}
+    detMap = {'ccd01': 'S29', 'ccd02': 'S30', 'ccd03': 'S31', 'ccd04': 'S25', 'ccd05': 'S26',
```

Comment:
Updated to use DecamMapper.detectorNames here and for detSerialMap (which is just a flip of key:value to value:key).  If the mapper does go away, this will suddenly fail, and will need to be handled at that time.
I think the "more generic" place is the obs_decam Instrument definition, but I'm not sure.